src/framework/midi/internal/platform/osx/coremidioutport.cpp

```diff
@@ -309,35 +297,60 @@ Ret CoreMidiOutPort::sendEvent(const Event& e)
 
     OSStatus result;
     MIDITimeStamp timeStamp = AudioGetCurrentHostTime();
+    static bool doLog = false; // kors::logger::Logger::instance()->isLevel(kors::logger::Level::Debug);
 
     if (__builtin_available(macOS 11.0, *)) {
-        MIDIProtocolID protocolId = configuration()->useMIDI20Output() ? m_core->destinationProtocolId : kMIDIProtocol_1_0;
+        MIDIProtocolID protocolId = kMIDIProtocol_2_0; // configuration()->useMIDI20Output() ? m_core->destinationProtocolId : kMIDIProtocol_1_0;
```

Comment:
What about this change? Looks like that removes the effect of the checkbox added in https://github.com/musescore/MuseScore/pull/12329. If your changes means that we won't need that checkbox anymore, then it would be better to remove it, rather than leaving it there doing nothing.
Also, looks like all the `destinationProtocolId` stuff becomes unused too.

Reply:
Apple says, the Core Midi framework translates between Midi 2.0 and Midi 1.0 messages behind the scenes.
So checking  the preferred protocol of the output port is optional and merely allows to filter out ahead what would be thrown away anyway. But I think it is cleaner not doing that.

Regarding the checkbox: Quote from #12329: A use case for this is the following: on macOS, the MIDI IAC Driver reports that it supports MIDI 2.0, but if you connect any MIDI 1.0 apps to that driver, those apps won't understand the MIDI 2.0 messages. In that case, you want to force MuseScore to produce MIDI 1.0 output even though the receiver reports that it supports MIDI 2.0.

If everything works like Apple says, then this should not be a problem. And from my testing my new code works with IAC Driver and Midi 1.0 app connected.

Leaving the checkbox in might be a good idea for some transitioning period for easily testing the two code paths (macOS 11 + midi 2.0 vs old macOS + midi 1.0 packet list) on a recent macOS. Or remove the checkbox but keep the setting option in a less prominent Debug-UI. What do you prefer? 

Anyway, I am happy to do the code changes that you suggested, but I would prefer to leave the UI changes to someone who better knows how to do that.